lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,538 @@
+      created_at: new Date(testGranule.createdAt),
+    }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateFileRecord handles nullable fields on source file data', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+
+  delete testFile.bucket;
+  delete testFile.checksum;
+  delete testFile.checksumType;
+  delete testFile.fileName;
+  delete testFile.key;
+  delete testFile.path;
+  delete testFile.size;
+  delete testFile.source;
+
+  await migrateGranuleRecord(testGranule, knex);
+  await migrateFileRecord(testFile, testGranule.granuleId, testGranule.collectionId, knex);
+
+  // Also unsure of condition for null bucket and key
+  const record = await filePgModel.get(knex, {});
+
+  t.deepEqual(
+    omit(record, fileOmitList),
+    {
+      bucket: null,
+      checksum_value: null,
+      checksum_type: null,
+      file_size: null,
+      file_name: null,
+      key: null,
+      source: null,
+      path: null,
+    }
+  );
+});
+
+test.serial('migrateGranulesAndFiles skips already migrated granule record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await Promise.all(testGranule.files.map((file) => s3Utils.s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: 'some-body',
+  })));
+  await migrateGranuleRecord(testGranule, knex);
+  await granulesModel.create(testGranule);
+
+  t.teardown(() => {
+    granulesModel.delete(testGranule);
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 0,
+      failed: 0,
+      skipped: 0,
+      success: 0,
+    },
+    granulesSummary: {
+      dynamoRecords: 1,
+      failed: 0,
+      skipped: 1,
+      success: 0,
+    },
+  });
+
+  const records = await knex(tableNames.granules);
+  t.is(records.length, 1);
+});
+
+test.serial('migrateGranulesAndFiles processes multiple granules and files', async (t) => {
+  const {
+    collectionPgModel,
+    executionPgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const alternateBucket = cryptoRandomString({ length: 10 });
+  await s3Utils.createBucket(alternateBucket);
+
+  const testExecution2 = fakeExecutionRecordFactory();
+  await executionPgModel.create(
+    knex,
+    testExecution2
+  );
+
+  const testCollection2 = fakeCollectionRecordFactory();
+  await collectionPgModel.create(
+    t.context.knex,
+    testCollection2
+  );
+
+  const testGranule1 = testGranule;
+  const testGranule2 = generateTestGranule(testCollection2, testExecution2.arn, alternateBucket);
```

Comment:
Is using a new collection, execution, and bucket necessary for this test?

Reply:
No. No it isn't 😞 [`5374077` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/53740775e8529f9099375c223a3b91bb42e27b0e)